net/udpxy/files/udpxy.init

```diff
@@ -54,11 +55,21 @@ start_instance() {
 	config_get_bool aux "$cfg" 'respawn' '0'
 	[ "$aux" = 1 ] && procd_set_param respawn
 
+	config_get dev "$cfg" "source"
+	[ -n "$dev" ] && procd_set_param netdev $dev
+
 	procd_close_instance
 }
 
-service_triggers() { 
-	procd_add_reload_trigger "udpxy" 
+service_triggers() {
+	local cfg
+	local dev
+	local ifname
+	procd_add_reload_trigger "udpxy"
+	cfg=$(config_foreach echo $1 udpxy)
+	config_get dev "$cfg" "source"
+	[ -n "$dev" ] && ifname=$(uci show network | grep $dev | awk '{split($0,arr,".");print arr[2]}')
```

Comment:
So this commit introduces a new UCI config variable `dev`. It's not explained in the commit message, why is it needed, how is the end user supposed to set it properly and when? Can't it be auto-detected?
This method of getting UCI variables is wrong, and also `grep | awk` pipe is suboptimal, and why `awk` where you can do the same with `cut`? But that doesn't matter much because the script isn't supposed to be using `uci show network` in the first place.

Reply:
> So this commit introduces a new UCI config variable `dev`. It's not explained in the commit message, why is it needed, how is the end user supposed to set it properly and when? Can't it be auto-detected? This method of getting UCI variables is wrong, and also `grep | awk` pipe is suboptimal, and why `awk` where you can do the same with `cut`? But that doesn't matter much because the script isn't supposed to be using `uci show network` in the first place.

variable `dev` and all other code does not related to end user, all those code is aiming at to restart the udpxy service when the corresponding interface status changed.